src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -129,33 +880,78 @@ public static IEnumerable<Diagnostic> GetEffectiveDiagnostics(IEnumerable<Diagno
         /// </summary>
         public static bool IsDiagnosticAnalyzerSuppressed(DiagnosticAnalyzer analyzer, CompilationOptions options, Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = null)
         {
-            if (analyzer == null)
-            {
-                throw new ArgumentNullException(nameof(analyzer));
-            }
+            VerifyAnalyzerArgumentForStaticApis(analyzer);
 
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
-            Action<Exception, DiagnosticAnalyzer, Diagnostic> voidHandler = (ex, a, diag) => { };
-            onAnalyzerException = onAnalyzerException ?? voidHandler;
             var analyzerExecutor = AnalyzerExecutor.CreateForSupportedDiagnostics(onAnalyzerException, AnalyzerManager.Instance);
-
             return AnalyzerDriver.IsDiagnosticAnalyzerSuppressed(analyzer, options, AnalyzerManager.Instance, analyzerExecutor);
         }
 
         /// <summary>
+        /// Create an analyzer exception diagnostic.
+        /// </summary>
+        public static Diagnostic CreateAnalyzerExceptionDiagnostic(DiagnosticAnalyzer analyzer, Exception exception)
```

Comment:
We can probably avoid this API if it is fine to clone the functionality in `AnalyzerExecutor.CreateAnalyzerExceptionDiagnostic` into the IDE driver. This API basically allows the host to report analyzer exception diagnostics for its own host specific analyzers/actions.